### ISSUE TITLE
Use user streams to retrieve tweets

### ISSUE THREAD

**Issue by bdonlan** [cezarsa/silver_bird#198]
Twitter just released a new user streaming API: http://dev.twitter.com/pages/user_streams
With this API, messages can be received immediately, without waiting for a poll (and consuming API hits). It would be nice if chromed bird could use this API to retrieve tweets.

However, since the twitter API does not use web sockets, but rather uses a normal, but long-running HTTP request, this will likely need additional support in chrome, a proxy server somewhere, a flash component, or something similar to make data available in real time to JavaScript. So consider this a long-term wishlist entry unless there's some obscure chrome extension API feature that would make this possible :)

**Comment by bdonlan** [comment]
Release announcement, since the API docs haven't been updated yet: http://groups.google.com/group/twitter-development-talk/browse_thread/thread/39f188e6f71d93b6?pli=1

**Comment by bdonlan** [comment]
http://github.com/r/twstreamer#readme <-- this may work ... it does say it's not production ready yet, though, and we'll need a fallback if flash is unavailable

**Comment by cezarsa** [comment]
Yeah, I've thought about that some time ago, I even started a discussion at the twitter-development mailing list asking whether there was a chance for streaming APIs to support web sockets connections in the future. I got no clear answer and was told about this twstreamer project. :)
IMHO, web sockets would make much more sense than persistent HTTP connections, the way the streaming API works today sounds a lot like a hack.

Anyway, coming back to the real world, using a web socket proxy is out of question. Server infrastructure is expensive (even using some cloud hosting like GAE [which I think started supporting websockets along with http]) and the money from donations (~$50 so far :) ) wouldn't be enough to cover these expenses.

I think the only real option is using this Flash bridge... Every time I hear the word Flash it gives me some chills, but anyway.
Chrome extensions API currently doesn't seem to provide us any other way to open connections than XHR with relaxed same origin policies, and XHR will only return after the response is complete, which means never for the streaming API.

I think I'll try using twstreamer soon, it'll be important to measure how it would affect CB's memory signature and performance.

**Comment by metbril** [comment]
+1 for using the streaming api. tweetdeck chrome app already does
